pkg/apb/ext_creds.go

```diff
@@ -0,0 +1,119 @@
+package apb
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"regexp"
+	"strings"
+	"time"
+
+	logging "github.com/op/go-logging"
+)
+
+// HACK ALERT!
+// A lot of the current approach to extracting credentials and monitoring
+// output is *very* experimental and error prone. Entire approach is going
+// to be thrown out and redone asap.
+
+func extractCredentials(
+	output []byte, log *logging.Logger,
+) (*ExtractedCredentials, error) {
+	log.Info("{%s}", string(output))
+
+	log.Debug("Calling getPodName")
+	podname, _ := getPodName(output, log)
+	log.Debug("Calling monitorOutput on " + podname)
+	bindout, _ := monitorOutput(podname)
+	log.Info(string(bindout))
+
+	var creds *ExtractedCredentials
+	var err error
+	creds, err = buildExtractedCredentials(bindout)
```

Comment:
this made it hard to test because the output given to the extractCredentials is only used to get the podname. then we run oc get logs and use THAT output. So we really need a way to mock out the runCommand to have it return testable output. For now, I tested buildExtractedCredentials separately.